### PR TITLE
Let the Login Button Be Replaced with a Link to Support More Auth Flows

### PR DESCRIPTION
**Pull request by Slowki**
**Type:** feature + unit tests

The following has been addressed in the PR:

*  Unit or Functional tests are included in the PR

**Description:**
This PR adds the ability for auth plugins to set a property which lets them override the default behavior of the login button and turn it into a link. The goal of this is to provide better support for OAuth, OpenID, SAML, etc.

Here's the changes in action:
output

I'll put up a PR in [flow-types](https://github.com/verdaccio/flow-types) which can be merged concurrently with this PR to update `IPluginAuth`.

### CODE REVIEW

**Comment by juanpicado** [comment]
So, the idea is `login_url` should be in companion of some sort of OAuth plugin or something like that?

File: src/lib/auth.js [hunk 1]
```diff
@@ -31,6 +33,16 @@ class Auth implements IAuth {
...
     this.plugins = this._loadPlugin(config);
     this._applyDefaultPlugins();
+
+    this.plugins.forEach((plugin) => {
+        if (plugin.login_url) {
+            if (this.login_url && this.login_url != plugin.login_url) {
+                throw new Error('conflicting auth plugins: multiple plugins tried to set different login urls');
+            }
+
+            this.login_url = plugin.login_url;
```
Comment (juanpicado) [inline src/lib/auth.js]: This should be part of the plugins dev documentation

**Comment by Slowki** [comment]
@juanpicado Yep, I've working on creating an [OAuth plugin](https://github.com/Slowki/verdaccio-oauth) and this feature is required to make it work.

**Comment by juanpicado** [comment]
Refers: 
https://github.com/verdaccio/verdaccio/issues/353

**Comment by juanpicado** [comment]
@Slowki perfect, let me know whether you need something else 👍

File: src/lib/auth.js (hunk 1 above)
Comment (Slowki) [reply src/lib/auth.js]: :+1:

**Comment by juanpicado** [comment]
@Slowki any update?

**Pull request closed without merging**